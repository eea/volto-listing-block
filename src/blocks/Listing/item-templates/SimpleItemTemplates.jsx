import cx from 'classnames';
import { ConditionalLink } from '@plone/volto/components';

import { getVoltoStyles } from '@eeacms/volto-listing-block/schema-utils';

const getStyles = (props) => {
  const { itemModel = {} } = props;
  const res = {};
  if (itemModel.maxDescription) {
    res[`max-${itemModel.maxDescription}-lines`] = true;
  }
  if (itemModel.maxTitle) {
    res[`title-max-${itemModel.maxTitle}-lines`] = true;
  }
  return res;
};

const BasicItem = (props) => {
  const { item, className, itemModel = {}, isEditMode = false } = props;
  //   const { hasImage, imageOnRightSide, hasDate, hasDescription } = itemModel;
  const styles = getStyles(props);

  console.log('sss', props);

  return (
    <div
      className={cx(
        'u-item listing-item simple-listing-item',
        getVoltoStyles(styles),
        className,
      )}
    >
      <div className={`wrapper `}>
        <div className="slot-top">
          <ConditionalLink item={item} condition={!isEditMode}>
            <div className="listing-body">
              <p className={'listing-header'}>
                {item.title ? item.title : item.id}
              </p>
            </div>
          </ConditionalLink>
        </div>
        <div className="slot-bottom">{item?.extra}</div>
      </div>
    </div>
  );
};

export const SimpleItemLayout = (props) => {
  return <BasicItem {...props} />;
};
