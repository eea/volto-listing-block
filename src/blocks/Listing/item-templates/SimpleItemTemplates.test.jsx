import React from 'react';
import renderer from 'react-test-renderer';
import { SimpleItemLayout } from './SimpleItemTemplates';

jest.mock('@plone/volto/components', () => ({
  ConditionalLink: () => <div>ConditionalLink</div>,
}));

describe('SimpleItemTemplates', () => {
  it('renders correctly', () => {
    const item = {
      title: 'Simple listing title',
      description: 'Simple listing description',
      extra: 'Simple listing extra',
    };

    const itemModel = {
      '@type': 'Document',
    };

    const component = renderer.create(
      <SimpleItemLayout
        item={item}
        itemModel={itemModel}
        className="my-class"
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
