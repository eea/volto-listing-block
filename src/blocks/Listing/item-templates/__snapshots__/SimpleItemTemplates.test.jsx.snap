// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleItemTemplates renders correctly 1`] = `
<div
  className="u-item listing-item simple-listing-item my-class"
>
  <div
    className="wrapper"
  >
    <div
      className="slot-top"
    >
      <div>
        ConditionalLink
      </div>
    </div>
    <div
      className="simple-item-meta"
    />
  </div>
</div>
`;
